src/network_manager.rs

```diff
@@ -29,6 +29,21 @@ pub struct ActiveConnections {
     pub connections: HashMap<PeerId, PeerConnection>,
     pub listeners: HashMap<SocketAddr, TransportType>,
 }
+
+impl ActiveConnections {
+    /// Check if an address has to be refused or not
```

Comment:
The comment makes me think that the function will return `true` in case of the address being refused which is the contrary following the usage you made in `tcp.rs`

```suggestion
    /// Check if a new connection from a specific address can be accepted or not
```